packages/amplify-cli/src/index.ts

```diff
@@ -116,14 +119,61 @@ export async function run() {
     }
 
     return exitCode;
-  } catch (e) {
+  } catch (error) {
     // ToDo: add logging to the core, and log execution errors using the unified core logging.
-    errorHandler(e);
-    if (e.message) {
-      print.error(e.message);
-    }
-    if (e.stack) {
-      print.info(e.stack);
+    errorHandler(error);
+
+    if (error.name === 'JSONValidationError') {
+      const jsonError = <JSONValidationError>error;
+      let printSummary = false;
+
+      print.error(error.message);
+
+      if (jsonError.unknownFlags?.length > 0) {
+        print.error('');
+        print.error(
+          `The following feature flags are defined in configuration are unknown by the currently running version of Amplify CLI:`,
+        );
+
+        for (const unknownFlag of jsonError.unknownFlags) {
+          print.error(`  - ${unknownFlag}`);
+        }
+
+        printSummary = true;
+      }
+
+      if (jsonError.otherErrors?.length > 0) {
+        print.error('');
+        print.error(`The following feature flags have validation errors:`);
+
+        for (const otherError of jsonError.otherErrors) {
+          print.error(`  - ${otherError}`);
+        }
+
+        printSummary = true;
+      }
+
+      if (printSummary) {
+        print.error('');
+        print.error(
+          `The project was pushed with a newer version of the Amplify CLI that is being used now. Update to the newer version that supports the feature flags shown as unknown above.`,
```

Comment:
This issue likely happens when the project has been pushed with a newer version of Amplify CLI. Try updating to a newer version of Amplify CLI by running "amplify upgrade"